bitmark-references/bitmark-cli/bitmark-cli.md

```diff
@@ -0,0 +1,729 @@
+# bitmark-cli Command Reference
+
+## Basic Command Structure
+
+~~~
+bitmark-cli [global-options] command [command-options]
+~~~
+
+## Global Options
+
+Options are listed as long form with any single character abbreviation.
+
+`--verbose|-v`
+
+: This allows the CLI to output additional information for each
+command.  Most commands will output the JSON request and response sent
+to the bitmarkd.
+
+`--network|-n  [bitmark|testing|local]`
+
+: Determines which network the command will be sent to from the following possible values:
+* `bitmark` the live network which uses live BTC or LTC to pay for the transactions.
+* `testing` a network for testing newly developed programs, it uses testnet coins to pay for transactions.
+* `local` a special case for running a regression test network on the loopback interface.
+
+`--connection|-c NUMBER`
+
+: Optionally allows for one of the other connections listed in the
+configuration file to be used instead of defaulting to the first one.
+
+`--identity|-i NAME`
+
+: Optionally override the default identity to allow running commands
+as another account.  Extra accounts are appended to the configuration
+file using the `add` command.
+
+`--password|-p PASSWORD`
+
+: This is primarily used for regression testing scripts to supply a
+password.  It is not recommended for normal use as the password will
+be left in the command history or be viewable by a process display
+command (top, ps, etc.)
+
+`--use-agent|-u EXECUTABLE`
+
+: This is the way to access a password stored in a password manager. the
+executable must accept the following arguments:
```

Comment:
I am not sure if I understand correctly, from description I thought following commands should be set in the order, e.g. first parameter is `confirm`, second parameter is `cache-id`?

Reply:
I always consider `--aaa=bbb` Or `-a` as an option and not as an argument